Fix spinner size at spot price

IYTEC opened the pull request:
# ℹ Overview
Fix spinner size at spot price
<!--- Provide an overview of the Pull Request -->

### 📝 Related Issues
- closes #827 
<!--- Pin any related issues -->

### 🔐 Acceptance:
<!-- Ensure the following are completed and mark the result with an [X] -->

- [ ] `yarn test` passes

atrefonas reviewed (changes requested):
The spinner is not small enough and doesn't fit into the bar - it causes the same bad visual effect before of the bar expanding and contracting.

atrefonas commented:
Just echoing my comment "This bar should be a constant height even while data is loading. Otherwise it flickers too much and makes the orderbook resize in a weird way when data loads quickly."

atrefonas reviewed (changes requested):
It's better now but the spot price spinner is not displaying while the page loads.

https://user-images.githubusercontent.com/72869083/197341828-0e3be675-3834-4893-9cb3-7e1e3731e8ff.mov

IYTEC commented:
> It's better now but the spot price spinner is not displaying while the page loads.
> 
>  spinner-not-visible.mov

It spins for me correctly here. Maybe the network is very fast 😄 
https://www.loom.com/share/1ea5e92b3b5a448d96a32b403a02ca6b

atrefonas commented:
In your video, the spot price isn’t spinning. The entire orderbook is
spinning.

On Sat, Oct 22, 2022 at 9:43 AM Ikechi ***@***.***> wrote:

> It's better now but the spot price spinner is not displaying while the
> page loads.
>
> spinner-not-visible.mov
>
> It spins for me correctly here. Maybe the network is very fast 😄
> https://www.loom.com/share/1ea5e92b3b5a448d96a32b403a02ca6b
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/algodex/algodex-react/pull/837#issuecomment-1287797505>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARL6JW4NNV3I5KQ5X46Q66TWEPVPLANCNFSM6AAAAAARLQTGWY>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

atrefonas commented:
> In your video, the spot price isn’t spinning. The entire orderbook is spinning.
> […](#)
> On Sat, Oct 22, 2022 at 9:43 AM Ikechi ***@***.***> wrote: It's better now but the spot price spinner is not displaying while the page loads. spinner-not-visible.mov It spins for me correctly here. Maybe the network is very fast 😄 https://www.loom.com/share/1ea5e92b3b5a448d96a32b403a02ca6b — Reply to this email directly, view it on GitHub <[#837 (comment)](https://github.com/algodex/algodex-react/pull/837#issuecomment-1287797505)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ARL6JW4NNV3I5KQ5X46Q66TWEPVPLANCNFSM6AAAAAARLQTGWY> . You are receiving this because your review was requested.Message ID: ***@***.***>

Oh nevermind, I do see the spot price spinner in your video.

Can you move the spinner to be horizontally centered?

atrefonas reviewed (changes requested):
The spinner is still not appearing when the price is blank. See the second half of this video.

By the way, in Google chrome you can slow down your network responses to help with debugging.

[ice_video_20221022-102738.webm](https://user-images.githubusercontent.com/72869083/197344604-9d37a71f-12e1-4840-86c8-0da681e6298d.webm)

atrefonas commented:
Also, when the price is blank, there should be a space between the "--" and the "0.00%". It looks like they are crunched together.

IYTEC commented:
I have this as the current build demo for me
https://www.loom.com/share/a6e684e2adc1493391789a4786ce0079

This is another build, where I remove the empty state and just show the spinner.
https://www.loom.com/share/fe68afa072f54425800f3468f4c61928

IYTEC commented:
I think the response is cached, I notice assets that have loaded does not show the spinner on second attempt.

IYTEC commented:
Depending on which ever we prefer, I can add back the empty state or fix the CI Jest failing

atrefonas commented:
I think the second build is better since it shows the spinner more. Can you implement that and fix the CI for it?

atrefonas reviewed (changes requested):
When it is loading a cached page (one that you already visited), it shouldn't show 0.00 for the price. It should either be -- or show a spinner.

atrefonas commented:
> When it is loading a cached page (one that you already visited), it shouldn't show 0.00 for the price. It should either be -- or show a spinner.

Loading spinner is preferable if the non-zero price is not cached or not available.

IYTEC committed:
Merge branch 'development' of https://github.com/algodex/algodex-react into issue-827-Fix_Spot_price_loading_spinner_should_be_smaller_and_fit_into_bar_on_desktop_

IYTEC committed:
Dash as replacement for empty spot price

IYTEC commented:
I removed the 0.00 showed at the beginning.

I have added the update such that, it shows the spinner when fetching new data, shows the dash (--) when price is not defined, and populate the view when data is available.

atrefonas reviewed (changes requested):
The loading spinner needs to show while the data is loading. If data is loading, it should not show "--". That should only show after the data has loaded and there is no price.

If it is showing a cached price, it should show that and not the loading spinner. However, it seems like the prices don't get cached.

https://user-images.githubusercontent.com/72869083/197785443-0fdaa165-49ad-4c89-bcce-52f6a27a0a85.mov

atrefonas commented:
> The loading spinner needs to show while the data is loading. If data is loading, it should not show "--". That should only show after the data has loaded and there is no price.
> 
> If it is showing a cached price, it should show that and not the loading spinner. However, it seems like the prices don't get cached.
> 
>  Screen.Recording.2022-10-25.at.9.24.38.AM.mov

0:08 - 0:12 shows the issue

IYTEC committed:
Merge branch 'development' of https://github.com/algodex/algodex-react into issue-827-Fix_Spot_price_loading_spinner_should_be_smaller_and_fit_into_bar_on_desktop_

IYTEC committed:
Update spinner view
